### DISCUSSION TITLE
Card with UnstyledButton set to underlying component

### DISCUSSION THREAD

**Discussion by sienkiewiczkm** [mantinedev/mantine#4684] [Q&A] [answered]
Hello!

I would like to build an interface where [`Card`](https://mantine.dev/core/card/) would act as a button (without style, so [`UnstyledButton`](https://mantine.dev/core/unstyled-button/) sounds perfect) with `onClick` handler attached. This definitely can work, but I cannot get the types right:

https://codesandbox.io/s/mantine-card-unstyledbutton-22yhx2?file=/src/App.tsx

If you look closely, `console.log` actually is triggered on click, but the typechecker complains:

```tsx
<Card
  withBorder
  shadow="lg"
  m="xl"
  maw={100}
  component={UnstyledButton}
  onClick={() => console.log("onClick!")}
>
  {/* ... */}
</Card>
```

![image](https://github.com/mantinedev/mantine/assets/1215472/4e882eec-fff9-42e1-96d6-acd1400f8637)

I will appreciate any help! Thank you!

**Comment by rtivital** [comment, marked as answer]
Use `<Card component="button"  />` instead of UnstyledButton. You cannot pass a polymorphic component in `component` prop of other `polymorphic` component.

**Reply by sienkiewiczkm** [reply]
Oh, that makes sense. Unfortunately this will apply default browser styles - which will become more obvious when `withBorder` and `shadow` is turned off - but I can live with some additional CSS. Thank you for such quick response.
